index.js

```diff
@@ -12,6 +12,22 @@ function makeDriver(opts) {
     this.instance = puppeteer;
 
     const autoScroll = async (page) => {
+      let timer;
+      let timeout = new Promise((resolve, reject) => {
+        timer = setTimeout(() => {
+          reject('Timed out');
+        }, 30000)
+      })
+
+      await Promise.race([
+        timeout,
+        page.waitForNavigation()
+      ]).catch(() => {
+        console.log('Timed out while waiting for navigation (handled)')
+      });
+
+      clearTimeout(timer);
```

Comment:
This is a workaround to broken navigation when some sites try to make some sort of redirections.
So waiting for navigation helps for those cases.